File: CodeForces/Contest/Round #528 (Div 2)/E528.java
//make sure to make new file!
import java.io.*;
import java.util.*;

public class E528{
   
   public static void main(String[] args)throws IOException{
      BufferedReader f = new BufferedReader(new InputStreamReader(System.in));
      PrintWriter out = new PrintWriter(System.out);
      
      int t = Integer.parseInt(f.readLine());
      
      for(int q = 0; q < t; q++){
      
         int n = Integer.parseInt(f.readLine());
      
         String s = f.readLine();
         String a = f.readLine();
         String b = f.readLine();
      
         HashSet<Character> hs = new HashSet<Character>();
         
         for(int k = 0; k < n; k++){
            hs.add(s.charAt(k));
         }
         
         HashMap<Character,Character> hm = new HashMap<Character,Character>();
         
         boolean solved = false;
         for(char c : hs){
            if(ctoi(c) == n-1){
               if(hm.containsKey(c)){
                  out.println("NO");
                  solved = true;
                  break;
               } else {
                  hm.put(c,c);
               }
            } else {
               if(hm.containsKey((char)(ctoi(c) + 1))){
                  if(hm.containsKey(c)){
                     out.println("NO");
                     solved = true;
                     break;
                  }
               } else {
                  hm.put(c,(char)(ctoi(c) + 1)));
               }
            }
         }
         if(solved) continue;
         
         String s2 = "";
         for(int k = 0; k < n; k++){
            if(hm.containsKey(s.charAt(k))){
               s2+=hm.get(s.charAt(k));
            } else {
               s2+=s.charAt(k);
            }
         }
         
         if(s2.compareTo(s) == 
         
         
      
      }
      
      out.close();
   }
   
   public int ctoi(char c){
      return (int)c - 97;
   }
   
}

File: CodeForces/Contest/Round #582 (Div 3)/E528.kt
//make sure to make new file!
import java.io.*
import java.util.*

object E528 {

    @Throws(IOException::class)
    fun main(args: Array<String>) {
        val f = BufferedReader(InputStreamReader(System.`in`))
        val out = PrintWriter(System.out)

        val n = Integer.parseInt(f.readLine())

        var a = f.readLine()
        var b = f.readLine()

        val each = HashSet<Character>()
        each.add('a')
        each.add('b')
        each.add('c')

        val doubles = HashSet<String>()
        doubles.add("aa")
        doubles.add("bb")
        doubles.add("cc")

        if (doubles.contains(b)) {
            val temp = b
            b = a
            a = temp
        }

        val sb = StringBuilder("")

        out.println("YES")
        if (n == 1) {
            out.println(find1(a, b))
        } else if (doubles.contains(b)) {
            //both are doubles
            //ex: aa, bb -> abababccc
            if (a.equals(b)) {
                val next = if (a.charAt(0) === 'a') 'b' else 'a'
                b = "" + next + next
            }
            for (k in 0 until n) {
                sb.append(a.charAt(0))
                sb.append(b.charAt(0))
            }
            each.remove(a.charAt(0))
            each.remove(b.charAt(0))
            val last = get(each)
            for (k in 0 until n) {
                sb.append(last)
            }

            out.println(sb.toString())
        } else if (doubles.contains(a)) {
            //one (a) is double

            if (b.charAt(0) === a.charAt(0) || b.charAt(1) === a.charAt(0)) {
                //ex: aa, ab -> acacacbbb
                val last = if (b.charAt(0) === a.charAt(0)) b.charAt(1) else b.charAt(0)
                each.remove(a.charAt(0))
                each.remove(last)
                val second = get(each)
                for (k in 0 until n) {
                    sb.append(a.charAt(0))
                    sb.append(second)
                }
                for (k in 0 until n) {
                    sb.append(last)
                }

                out.println(sb.toString())
            } else {
                //ex: aa, bc -> acacacbbb

                for (k in 0 until n) {
                    sb.append(a.charAt(0))
                    sb.append(b.charAt(1))
                }
                for (k in 0 until n) {
                    sb.append(b.charAt(0))
                }

                out.println(sb.toString())
            }
        } else {
            //no doubles
            //ex: ab, bc -> b,a,c -> bbbaaaccc

            val f1 = find1(a, b)

            for (k in 0..2) for (j in 0 until n) sb.append(f1.charAt(k))
            out.println(sb.toString())


        }






        out.close()
    }

    operator fun get(hs: HashSet<Character>): Char {
        for (c in hs) return c
        return 'a'
    }

    fun find1(a: String, b: String): String {
        for (k in 97..99) {
            for (j in 97..99) {
                for (h in 97..99) {
                    if (k == j || j == h || h == k)
                        continue
                    val s1 = "" + k.toChar() + j.toChar()
                    val s2 = "" + j.toChar() + h.toChar()

                    if (s1.equals(a) || s1.equals(b))
                        continue
                    if (s2.equals(a) || s2.equals(b))
                        continue

                    return s1 + h.toChar()
                }
            }
        }

        while (true);
        //return "big sad";
    }


}